lib/jira_api/template_handler.py

```diff
@@ -0,0 +1,158 @@
+    An jira issue description is a string made up of one or more rendered jinja2 templates.
+    """
+
+    # Holds absolute filepath to jira template metadata (from current dir)
+    # .../st2-cloud-pack/lib/jira_api/jira_templates.yaml
+    JIRA_TEMPLATE_METADATA_FP = (
+        Path(__file__).resolve().parent / "jira_template_schemas.yaml"
+    )
+
+    # Holds absolute dirpath to directory where jira template files are stores
+    # .../st2-cloud-pack/jira_templates
+    JIRA_TEMPLATE_ROOT_DIR = (
+        Path(__file__).resolve().parent.parent.parent / "jira_templates"
+    )
+
+    def __init__(self, template_metadata=None):
+        self._template_env = Environment(
+            loader=FileSystemLoader(self.JIRA_TEMPLATE_ROOT_DIR)
+        )
+
+        self._template_metadata = (
+            template_metadata
+            if template_metadata
+            else self._load_all_metadata(self.JIRA_TEMPLATE_METADATA_FP)
+        )
+
+    @staticmethod
+    def _load_all_metadata(metadata_fp) -> Dict:
+        """
+        Method which reads in jira templates yaml file. This file holds metadata for each template name.
+        """
+        with open(metadata_fp, "r", encoding="utf-8") as stream:
+            try:
+                return safe_load(stream)
+            except YAMLError as exc:
+                raise JiraTemplateError(
+                    "could not load template metadata file"
+                ) from exc
+
+    @staticmethod
+    def _parse_template_attrs(
+        template_details: JiraTemplateDetails, template_schema: Dict[str, str]
+    ) -> Dict[str, str]:
+        """
+        Static method which returns values for all attributes to replace for the template that has been provided.
+        attribute values either set to mapped values in 'given_vals' or set to a default given by 'template_schema'
+        :param template_details: A dataclass containing info on templates
+        :param template_schema: A dictionary holding required attribute names mapped to defaults
+        """
+        template_params = {}
+        if template_details.template_params:
+            template_params = template_details.template_params
+
+        given_vals = {
+            key: val for key, val in template_params.items() if val is not None
+        }
+        attrs = {
+            key: given_vals.get(key, default)
+            for key, default in template_schema.items()
+        }
+
+        missing_attrs = [key for key, val in attrs.items() if val is None]
+        if missing_attrs:
+            raise JiraTemplateError(
+                f"Missing attributes required for template {template_details.template_name}: {', '.join(missing_attrs)}"
+            )
+
+        return attrs
+
+    def _get_template_file(self, template_fp) -> Template:
+        """
+        Method to read in template file as a jinja template from filepath.
+        :param template_fp: a relative filepath for template file starting from 'JIRA_TEMPLATE_ROOT_DIR'
```

Comment:
Could you add what this method returns in the docstring - what kind of `Template` is being returned?